File: src/app/functions/Worker.java
package app.functions;

import java.util.ArrayList;

public class Worker {
  private Thread worker;
  private ArrayList<Runnable> args = new ArrayList<>();

  public synchronized void threadedWorker(Runnable r) {
    worker = new Thread(r);
    worker.start();
  }

  public synchronized void killWorker() {
    worker.interrupt();
  }

  public synchronized void groupedWorker(Runnable... r) {
    for (Runnable r1 : r) {
      worker = new Thread(r1);
      worker.start();
      args.add(r1);
    }
  }

  public synchronized void killSpecWorker(int index) {
    ((Thread) args.get(index)).interrupt();
  }
}

File: kt/app/functions/Worker.kt
package app.functions

import java.lang.Runnable
import java.util.ArrayList
import kotlin.jvm.Synchronized

/**
 * @author Jack Meng
 */
class Worker {
    private var worker: Thread? = null
    private val args = ArrayList<Runnable>()
    @Synchronized
    fun threadedWorker(r: Runnable?) {
        worker = Thread(r)
        worker!!.start()
    }

    @Synchronized
    fun killWorker() {
        worker!!.interrupt()
    }

    @Synchronized
    fun groupedWorker(vararg r: Runnable) {
        for (r1 in r) {
            worker = Thread(r1)
            worker!!.start()
            args.add(r1)
        }
    }

    @Synchronized
    fun killSpecWorker(index: Int) {
        (args[index] as Thread).interrupt()
    }
}